django/contrib/admin/static/admin/js/admin/DateTimeShortcuts.js

```diff
@@ -406,6 +403,14 @@
             DateTimeShortcuts.calendarInputs[num].value = d.strftime(get_format('DATE_INPUT_FORMATS')[0]);
             DateTimeShortcuts.calendarInputs[num].focus();
             DateTimeShortcuts.dismissCalendar(num);
+        },
+        escapeFormat: function(format) {
+            format = format.replace(/\\/g, "\\\\")
+                .replace(new RegExp('\r', 'g'), '\\r')
+                .replace(new RegExp('\n', 'g'), '\\n')
+                .replace(new RegExp('\t', 'g'), '\\t')
+                .replace(new RegExp("'", 'g'), "\\'");
```

Comment:
Do we need to create a new `RegExp` objects? `/\t/g` syntax works for me:
```suggestion
            format = format.replace(/\\/g, "\\\\")
                .replace(/\r/g, "\\r")
                .replace(/\n/g, "\\n")
                .replace(/\t/g, "\\t")
                .replace(/'/g, "\\'");
```